### PR TITLE
Megatron LLM documentation updates

### PR DESCRIPTION
**Pull request by ssh-meister**
# What does this PR do ?

- Added description of supported positional embedding types in GPT and T5 models 
- Added description of positional interpolation
- Added information about Flash attention

### CODE REVIEW

File: docs/source/nlp/nemo_megatron/flash_attention.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+   model.use_flash_attention=True
+
+T5
+^^
+To enable Flash Attention while Megatron T5 model training, modify the following configuration: 
+
+.. code::
+
+   model.use_flash_attention=True
```
Comment (jubick1337) [inline docs/source/nlp/nemo_megatron/flash_attention.rst:19]: don't we need to set it both in encoder and decoder separately?
@ssh-meister @hsiehjackson

File: docs/source/nlp/nemo_megatron/positional_embeddings.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+------------------------
+`Position Interpolation (PI) <https://arxiv.org/pdf/2306.15595.pdf>`_  is a method introduced to extend the context window sizes of Rotary Position Embedding (RoPE)-based pretrained large language models (LLMs). The central principle of PI is to reduce the position indices so that they align with the initial context window size through interpolation.
+
+Positional Interpolation is supported in Megatron GPT SFT models. Set RoPE Interpolation factor for sequence length :code:`seq_len_interpolation_factor` to enable it.  
+
+.. code::
+   
+   model.position_embedding_type='rope'
+   model.rotary_percentage=1.0
+   model.seq_len_interpolation_factor: null
```
Comment (jubick1337) [inline docs/source/nlp/nemo_megatron/positional_embeddings.rst:103]: why null?

Comment (ssh-meister) [reply docs/source/nlp/nemo_megatron/positional_embeddings.rst]: @jubick1337, thank you! Changed to 2

File: docs/source/nlp/nemo_megatron/flash_attention.rst (hunk 1 above)
Comment (ssh-meister) [reply docs/source/nlp/nemo_megatron/flash_attention.rst]: Thank you! Fixed!

**Comment by jubick1337** [comment]
Could you add links to the new pages to the sidebar?
Otherwise, there's no way to access it.

**Review by ekmb — APPROVED** [review]
Thanks!

**Pull request merged**